lib/spack/spack/solver/concretize.lp

```diff
@@ -68,18 +67,18 @@ unification_set("generic_build", node(X, Child))
      not multiple_unification_sets(Child),
      unification_set(_, ParentNode).
 
-% Any dependency of type "build" in a unification set that is in the leaf unification set,
-% stays in that unification set
-unification_set(SetID, ChildNode)
-  :- attr("depends_on", ParentNode, ChildNode, Type),
-     Type == "build",
-     SetID == "generic_build",
-     unification_set(SetID, ParentNode).
-
 unification_set(SetID, VirtualNode)
   :- provider(PackageNode, VirtualNode),
      unification_set(SetID, PackageNode).
 
+% Do not allow split dependencies, for now. This ensures that we don't construct graphs where e.g.
+% a python extension depends on setuptools@63.4 as a run dependency, but uses e.g. setuptools@68
+% as a build dependency.
+%
+% We'll need to relax the rule once we get to actual cross-compilation
```

Comment:
Maybe even before that? gcc runtime libs depend on glibc@x, compiler binaries on glibc@y but same architecture